pkg/trace/config/apply.go

```diff
@@ -245,15 +246,15 @@ func (c *AgentConfig) applyDatadogConfig() error {
 	if config.Datadog.IsSet("apm_config.max_memory") {
 		c.MaxMemory = config.Datadog.GetFloat64("apm_config.max_memory")
 	}
-	if config.Datadog.IsSet("apm_config.max_connections") {
-		c.MaxConnections = config.Datadog.GetInt("apm_config.max_connections")
-	}
 
 	// undocumented
 	c.ServiceWriterConfig = readServiceWriterConfigYaml()
 	c.StatsWriterConfig = readStatsWriterConfigYaml()
 	c.TraceWriterConfig = readTraceWriterConfigYaml()
 
+	// allow an extra 10% of the maximum number of connections for parallel writes to the Datadog API
+	c.TraceWriterConfig.SenderConfig.MaxConnections = int(math.Max(1, float64(c.ConnectionLimit)/10))
```

Comment:
So this defaults to 200? That's reasonable and should be more than enough even for running the agent under high load. Even if payload sends take 1s each, that gives us ~ 200 payloads/s.

Reply:
I don't know about the coupling of two independent variables, here `SenderConfig.MaxConnections` with  `c.ConnectionLimit`. I'd rather see them independant, even if one is constant / pre-computed based on something else.

Reply:
Also, I thought `max_connections` was deprecated (as too generic / not well defined) ; it could have been the time to give these configurations a proper meaning/scope?

Reply:
Yeah maybe they should be independent. It kind of makes sense for "max outgoing connections" to be derived from the "max incoming connections" (at least by default) because outgoing should generally be some small fraction of incoming.

Reply:
> Also, I thought max_connections was deprecated (as too generic / not well defined) ; it could have been the time to give these configurations a proper meaning/scope?

@LotharSee I'm not sure what you mean here. `max_connections` _is_ deprecated. It was just left over in some places, from where I've removed it in this PR. The `MaxConnections` you see on this line is not user configurable, is new, and refers strictly to the maximum number of allowed concurrent flushes in the trace writer.

> I don't know about the coupling of two independent variables, here SenderConfig.MaxConnections with c.ConnectionLimit. I'd rather see them independant, even if one is constant / pre-computed based on something else more related (actual throughput, maxTPS, num CPU, ...).

I'd find it hard to make it based on maxTPS or throughput as that would be dynamic. What woul be the point? These variables are already strong determining factors of how many concurrent flushes you'll have. I think in 99.9% of cases the max. (200) will _never_ be reached. I've tested with a pretty high throughput and never had more than 5-15 concurrent routines active. Consider I'm in the EU and posting to the US (does that matter?).

I have also considered using num. CPU but it has little meaning here. The more determining factor could be memory or maximum allowed open files (which is 1000 by default on most systems). So, 200 by default here in my opinion should be good for now. 

Let me know if you have more thoughts here and we can consider changes.

Personally I think this fits well into `connection_limit`.